app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,60 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all
+    @task = Task.new
+  end
+
+  def show
+    @task = Task.find(params[:id])
+    person_id = @task.person_id
+    @person = Person.find(person_id)
+  end
+
+  def new
+    @task = Task.new
+    @form_url = "/tasks"
+    @method = :post
+  end
+
+  def create
+    @task = Task.create(task_params)
+    if @task.valid?
+      redirect_to :root
+    else
+      render :new
+    end
+  end
+
+  def delete
+    @task = Task.find(params[:id])
+    @task.delete
+    redirect_to :root
+  end
+
+  def update_form
+    @task = Task.find(params[:id])
+    @form_url = "/update/#{@task.id}"
+    @method = :patch
+  end 
+
+  def update
+    @task = Task.find(params[:id])
+    @task.update(task_params)
+    redirect_to "/task/#{@task.id}"
+  end
+
+  def completed
+    @task = Task.find(params[:id])
+    @task.date_completed = Time.now
```

Comment:
You can collapse these three lines into one `update` method call in which you can set these two field values